server/operations/src/main/java/org/kaaproject/kaa/server/operations/service/akka/actors/core/EndpointActorMessageProcessor.java

```diff
@@ -298,6 +273,93 @@ private void sync(ActorContext context, SyncRequestMessage requestMessage) {
         }
     }
 
+    private SyncContext sync(ClientSync request) throws GetDeltaException {
+        if (!request.isValid()) {
+            LOG.warn("[{}] Request is not valid. It does not contain profile information!", endpointKey);
+            return SyncContext.failure(request.getRequestId());
+        }
+        SyncContext context = new SyncContext(new ServerSync());
+        context.setEndpointProfile(state.getProfile());
+        context.setRequestId(request.getRequestId());
+        context.setStatus(SyncStatus.SUCCESS);
+        context.setEndpointKey(endpointKey);
+        context.setRequestHash(request.hashCode());
+        context.setMetaData(request.getClientSyncMetaData());
+
+        LOG.trace("[{}][{}] processing sync. Request: {}", endpointKey, context.getRequestHash(), request);
+
+        context = operationsService.syncProfile(context, request.getProfileSync());
+
+        state.setProfile(context.getEndpointProfile());
+
+        if (context.getStatus() != SyncStatus.SUCCESS) {
```

Comment:
Yes, there might be profile resync which means that we don't need to process anything else